### PR TITLE
Add built-in video preview

### LINKED ISSUE #159106: Include built-in preview for audio and video
From #156558

Now that we support additional codecs, we should explore shipping a builtin extension for previewing audio and video, similar to our built-in image preview extension

### PR DESCRIPTION
**Pull request by mjbvz**
Fixes #159106

Also hooks up our service worker to support seeking in local video resources. This requires handling range requests properly

### CODE REVIEW

**Review by joaomoreno — APPROVED** [review]

File: extensions/image-preview/media/audioPreview.js [hunk 1]
```diff
@@ -30,30 +30,38 @@
 	container.className = 'audio-container';
 	document.body.appendChild(container);
 
-	const audio = new Audio(settings.src);
+	const audio = new Audio(settings.src === null ? undefined : settings.src);
```
Comment (ProKashif) [inline extensions/image-preview/media/audioPreview.js:33]:
```suggestion
	const audio = new Audio(settings?.src);
```

File: extensions/image-preview/src/audioPreview.ts [hunk 2]
```diff
@@ -92,11 +90,12 @@ class AudioPreview extends MediaPreview {
...
 
-	private async getResourcePath(webviewEditor: vscode.WebviewPanel, resource: vscode.Uri, version: string): Promise<string> {
+	private async getResourcePath(webviewEditor: vscode.WebviewPanel, resource: vscode.Uri, version: string): Promise<string | null> {
 		if (resource.scheme === 'git') {
 			const stat = await vscode.workspace.fs.stat(resource);
 			if (stat.size === 0) {
-				return this.emptyAudioDataUri;
+				// The file is stored on git lfs
+				return null;
 			}
```
Comment (ProKashif) [inline extensions/image-preview/src/audioPreview.ts:99]: stat.size === 0 return;

File: extensions/image-preview/src/videoPreview.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+</html>`;
+	}
+
+	private async getResourcePath(webviewEditor: vscode.WebviewPanel, resource: vscode.Uri, version: string): Promise<string | null> {
+		if (resource.scheme === 'git') {
+			const stat = await vscode.workspace.fs.stat(resource);
+			if (stat.size === 0) {
+				// The file is stored on git lfs
+				return null;
+			}
```
Comment (ProKashif) [inline extensions/image-preview/src/videoPreview.ts:99]:
```suggestion
				// The file is stored on git lfs
				stat.size === 0 && return;
```

File: extensions/image-preview/src/videoPreview.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+		}
+
+		// Avoid adding cache busting if there is already a query string
+		if (resource.query) {
+			return webviewEditor.webview.asWebviewUri(resource).toString();
+		}
```
Comment (ProKashif) [inline extensions/image-preview/src/videoPreview.ts:105]:
```suggestion
			resource.query && return webviewEditor.webview.asWebviewUri(resource).toString();
```

**Review by ProKashif — CHANGES_REQUESTED** [review]

**Comment by rbadapanda** [comment]
Could you please add a settings for 'autoplay' ?

**Pull request merged**